server/WFCORE-4291_restore_legacy_not_graceful_startup_mode.adoc

```diff
@@ -0,0 +1,75 @@
+
+One example scenario is this: if Deployment A depends -- at startup -- on external configuration, fetched by an HTTP request to Deployment B, Deployment A will fail to start as the HTTP request for that configuration will be rejected by the starting server.
+
+This RFE proposes a new, non-graceful startup mode that will allow requests to be handled much earlier in the application server boot process. In standalone mode, the system administrator can choose to enable this startup mode for that server, and in domain mode, the server group can be configured similarly, applying the configuration choice to all servers in that domain. Affected subsystems include Undertow, mod_cluster, EJB, batch, transactions, and concurrency.
+
+== Issue Metadata
+
+=== Issue
+* https://issues.jboss.org/browse/WFCORE-4291[WFCORE-4291]
+
+=== Related Issues
+* https://issues.redhat.com/browse/EAP7-1320[EAP7-1320]
+
+=== Dev Contacts
+* mailto:{email}[{author}]
+
+=== QE Contacts
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+* [X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* For standalone mode, this option will be exposed as a command line option only, with the addition of a `--graceful-startup=false` option. Since the value is needed before standalone.xml is parsed, it must come from the command line. Given this lack of start mode persistence, system administrators managing standalone instances will need to update their system scripts (`/etc/init.d`, `sysvinit`, `systemd`, etc.)
+
+* For domain mode, the configuration will be extended to allow the system administrator to configure how the Host Contoller will start the servers in the group. The `server-group` element will support a new `graceful-startup` attribute, whose default value will be `true`. Every server in the group will be configured to start in the same manner. 
+
+* Mixed domains must continue to work, of course, so a transformer must be added to the system that, for starting servers on older versions of Wildfly, discards the `graceful-startup` attribute when the value is set to the default (`true`), or rejects it otherwise, preventing the older server from starting. This will require the administrator to make an explicit decision about non-graceful startups in a mixed domain environment. The requirement that that Domain Controller be at least the same version, if not newer, than the managed Host Controllers remains unchanged. _NOTE: Using this feature in a mixed domain with hosts running EAP 6.x or older is not supported._
+
+* Servers restarted or reloaded will restart/reload in the same in which they were originally started: If a non-graceful start was requested, a non-graceful start will occur upon restart and reload.
+
+* If the administrator requests a non-graceful startup, but also specifies `start-mode=suspend`, the non-graceful startup request will be disregarded and the server(s) will start suspended.
```

Comment:
@jasondlee This also applies for the `reload` and `restart` operations?

Reply:
Yes. That's covered, i think, in the first sentence above. Is that insufficient or unclear?

Reply:
Well, `reload` operation is not a `restart` so having only the `startup` keyword here seems little bit confusing to me.

Reply:
> I'm confused: "Servers restarted or reloaded will restart/reload in the same in which they were originally started: If a non-graceful start was requested, a non-graceful start will occur upon restart and reload."

Ah, I'm sorry. Just to be clear - "non-graceful start" includes `reload` too, right?

Reply:
Yes.